speaker/layer2_controller.go

```diff
@@ -122,11 +117,13 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	})
 
 	// Are we first in the list? If so, we win and should announce.
-	if len(availableNodes) > 0 && availableNodes[0] == c.myNode {
+	owner := availableNodes[0] // we have returned if len()==0
```

Comment:
can you split the refactoring bits in a separate commit so the actual change is easier to follow?

Reply:
that small refactor took place to allow me to log a message which will be usefull  if someone runs memberlist disabled, with speaker with node selector.  
I would find better as part of this commit but I can split it also (or drop it if you think is not important)

Reply:
Fair enough.
Let's remove the comment though

Reply:
reverted